src/ekka_autoheal.erl

```diff
@@ -122,20 +134,21 @@ compare_view({Running1, _} , {Running2, _}) ->
 coordinator([{Nodes, _} | _]) ->
     ekka_membership:coordinator(Nodes).
 
--spec heal_partition(list()) -> list(node()).
-heal_partition([]) ->
-    [];
-%% All nodes connected.
-heal_partition([{_, []}]) ->
-    [];
-%% Partial partitions happened.
-heal_partition([{Nodes, []}|_]) ->
+heal_partition([{Nodes, []} | _] = SplitView) ->
+    %% Symmetric partition.
+    ?LOG(info, "Healing partition: ~p", [SplitView]),
     reboot_minority(Nodes -- [node()]);
-heal_partition([{Majority, Minority}, {Minority, Majority}]) ->
+heal_partition([{Majority, Minority}, {Minority, Majority}] = SplitView) ->
+    %% Symmetric partition.
+    ?LOG(info, "Healing partition: ~p", [SplitView]),
     reboot_minority(Minority);
-heal_partition(SplitView) ->
-    ?LOG(critical, "Cannot heal the partitions: ~p", [SplitView]),
-    error({unknown_splitview, SplitView}).
+heal_partition([{_Nodes, Stopped} | _Asymmetric] = SplitView) ->
```

Comment:
In this case, the safest is perhaps to reboot ALL nodes.

Reply:
I suspect the end result may be the same, each node will get `{inconsistent_database, starting_partitioned_network, OtherNode}` once Mnesia restarted (at least I've been seeing this events in the test logs).

https://www.erlang.org/doc/apps/mnesia/mnesia_chap7.html#recovery-from-communication-failure